mailet/api/src/main/java/org/apache/mailet/Mail.java

```diff
@@ -232,9 +245,12 @@ default boolean hasSender() {
      *
      * @return an Iterator (of Strings) over all attribute names
      * @since Mailet API v2.1
+     * @deprecated see {@link #attributeNames()}
      */
     Iterator<String> getAttributeNames();
 
+    Iterator<AttributeName> attributeNames();
```

Comment:
Why not a `Stream<AttributeName>`, which is more convenient for the consumer...

Reply:
Add the java doc explaining what this new method does

Reply:
Because it's a drop-in replacement, I'll have a look in the main PR to see if we could benefit from it.

Reply:
We can benefit from it: I'll change that.